File: numbers/src/main/java/org/dicio/numbers/util/ResourceOpener.java
package org.dicio.numbers.util;

import java.io.FileNotFoundException;
import java.io.InputStream;

public class ResourceOpener {
    public static InputStream getResourceAsStream(final String path) throws FileNotFoundException {
        final String normalizedPath = path.startsWith("/") ? path : "/" + path;
        final InputStream inputStream = ResourceOpener.class.getResourceAsStream(normalizedPath);

        if (inputStream == null) {
            throw new FileNotFoundException();
        }
        return inputStream;
    }
}

File: numbers/src/main/java/org/dicio/numbers/util/ResourceOpener.kt
package org.dicio.numbers.util

import java.io.FileNotFoundException
import java.io.InputStream

object ResourceOpener {
    @JvmStatic
    @Throws(FileNotFoundException::class)
    fun getResourceAsStream(path: String): InputStream {
        val normalizedPath = if (path.startsWith("/")) path else "/$path"
        val inputStream = ResourceOpener::class.java.getResourceAsStream(normalizedPath)
            ?: throw FileNotFoundException()

        return inputStream
    }
}
